### DISCUSSION TITLE
Running our own Free PeerJS Server

### DISCUSSION THREAD

**Discussion by JAA17** [peers/peerjs#1225] [Q&A]
Now the Free Heroku button has gone, can you please give a tutorial for us on how to set up our own permanent Free PeerJS Server? Where can we go?

Will the button to set up the GitPod PeerJS Server be a permanent server which is always on?

Once I click the **Open in Gitpod** button on the page:

https://github.com/peers/peerjs-server

t sets it all up fine. Also, when I type in the command **npm install peer -g** in the GitPod terminal, eveything appears to work and install.

But when I type **peerjs --port 9000 --key peerjs --path /myapp** it just says **Error: Error: listen EAFNOSUPPORT: address family not supported :::9000** and freezes.

How can I get this to work?

Also how can I get this to be always on, even when I log out of GitHub?

**Comment by jonasgloning** [comment]
Gitpod is only for development, not a permanent hosting solution.

I would recommend the public server. If your project outgrows the public server, you will have outgrown any free hosting.

That said, here are some [Heroku alternatives with a free tier](https://github.com/Engagespot/heroku-free-alternatives); maybe there is something for you – but I have no experience with any of them. I can also recommend Oracle Cloud with its very generous permanent free tier, but it’s a bit more complicated to set up and maintain.

**Comment by jonasgloning** [comment]
The master branch is now deployed to [fly.io](https://fly.io) using this [config file](https://github.com/peers/peerjs-server/blob/master/fly.toml) and this [GitHub Actions file](https://github.com/peers/peerjs-server/blob/master/.github/workflows/fly.yml).
Fly has a permanent free tier.

You can deploy your own server:
1. Create a new project directory
2. Create a `Dockerfile` in it that contains:
```Dockerfile
FROM peerjs/peerjs-server:latest
```
3. [Install the fly.io cli](https://fly.io/docs/hands-on/install-flyctl/)
4. Sign up using `fly auth signup`
5. [Create a project](https://fly.io/docs/languages-and-frameworks/dockerfile/) using `fly launch`
6. Change the `internal_port` property in `fly.toml` to `9000`

**Reply by ga676005** [reply]
I visit their [website](https://fly.io/docs/about/pricing/#plans) and see We don’t offer a “free tier.”

**Reply by jonasgloning** [reply]
If you read a bit further: Instead, we offer some free resource allowances that apply to all plans, including the Hobby plan, and includes enough usage per month to run a small full-stack app, full-time, _for free._
